libparsec/crates/client/src/certificates_ops/mod.rs

```diff
@@ -147,4 +147,8 @@ impl CertificatesOps {
             Err(err @ storage::GetCertificateError::Internal(_)) => Err(err.into()),
         }
     }
+
+    pub async fn get_current_self_profile(&self) -> anyhow::Result<UserProfile> {
+        self.storage.write().await.get_current_self_profile().await
```

Comment:
Why do we need to write to retrieve the current profile ?

Reply:
Because there is a cache

Reply:
Do you mean that we need to write to force the cache to be updated and have the latest `UserProfile` ?

Reply:
It's a `RwLock`, we don't `write` there

Reply:
Why do we need to `write` instead of `read` ?

Reply:
I understand after ready the code of `store.get_current_self_profile()`